src/api/cvc4cpp.h

```diff
@@ -2254,6 +2256,129 @@ struct CVC4_EXPORT RoundingModeHashFunction
   inline size_t operator()(const RoundingMode& rm) const;
 };
 
+/* -------------------------------------------------------------------------- */
+/* Statistics                                                                 */
+/* -------------------------------------------------------------------------- */
+
+/**
+ * Represents a snapshot of a single statistic value.
+ * A value can be of type int64_t, double, std::string or a histogram
+ * (`std::map<std::string, uint64_t>`).
+ * The value type can be queried (using `isInt`, `isString`, etc.) and
+ * the stored value can be accessed (using `getInt`, `getString`, etc.).
+ */
+class Stat
+{
+  struct StatData;
+
+ public:
+  friend class Statistics;
+  friend std::ostream& operator<<(std::ostream& os, const Stat& sv);
+  using HistogramData = std::map<std::string, uint64_t>;
+  /** Create from the given value. */
+  Stat() = delete;
+  Stat(const Stat& s);
+  ~Stat();
+  Stat& operator=(const Stat& s);
+
+  /** Is this value intended for experts only? */
+  bool isExpert() const;
+  /** Does this value hold a non-default value? */
+  bool hasValue() const;
+
+  /** Is this value an integer? */
+  bool isInt() const;
+  /** Return the integer value */
+  int64_t getInt() const;
+  /** Is this value a double? */
+  bool isDouble() const;
+  /** Return the double value */
+  double getDouble() const;
+  /** Is this value an string? */
+  bool isString() const;
+  /** Return the string value */
+  const std::string& getString() const;
+  /** Is this value an histogram? */
+  bool isHistogram() const;
+  /** Return the histogram value */
+  const HistogramData& getHistogram() const;
+
+ private:
+  Stat(bool expert);
+  Stat(bool expert, StatData&& sd);
+  /** Whether this statistic is only meant for experts */
+  bool d_expert;
+  std::unique_ptr<StatData> d_data;
+};
+
+std::ostream& operator<<(std::ostream& os, const Stat& sv);
+
+/**
+ * Represents a snapshot of the solver statistics.
+ * Once obtained, an instance of this class is independent of the `Solver`
+ * object: it will not change when the solvers internal statistics do, it
+ * will not be invalidated if the solver is destroyed.
+ * Statistics are generally categorized as public and expert statistics.
+ * Iterating on this class (via `begin()` and `end()`) shows only public
+ * statistics. Expert statistics can be accessed via iteration using
+ * `begin_all()` and `end_all()`, or by querying a single statistic by
```

Comment:
I think this is outdated, I don't see `begin_all()` or `end_all()`.

Reply:
Good catch!